### ISSUE TITLE
[iOS] VoiceOver no longer focuses on the back button when navigating to a new page

### ISSUE THREAD

**Issue by mehmetf** [flutter/flutter#62873]
b/162794400 

On iOS, it seems like voiceover no longer focuses on the back button when we navigate to a new page. Wondering if this is a possible regression?

Attached a demo video at head, and another demo video in our July 26th build where the focus is correct. (See internal bug for videos).

**Comment by chunhtai** [comment]
I added a comment in the internal bug. In short, this is a desired behavior introduced in https://github.com/flutter/engine/pull/19826, and developers should be able to choose which element to focus first (while we can't before this fix). I have to look at the code in order to figure out why it focus the button first.

**Comment by jefflim-google** [comment]
This is contrary to what central accessibility and flutter has guided us on in the past. Will ping you separately.

**Comment by chunhtai** [comment]
talked to @jefflim-google offline, and we agree to keep the api as is. I will close this issue as WAI because the fix should be on the application side. feel free to reopen if there is any other concern
